go/my-app/semconv/v0.1.0/my_app_custom_elements_total.go

```diff
@@ -0,0 +1,59 @@
+
+// Copyright 2025 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated from semantic convention specification. DO NOT EDIT.
+
+package semconv // TODO(bwplotka): Use id prefix or something more unique?
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type CustomElementsCategory string
+
+const (
+		ACustomElementsCategory CustomElementsCategory = "a"
+		BCustomElementsCategory CustomElementsCategory = "b"
+		OtherCustomElementsCategory CustomElementsCategory = "other"
+)
+
+func MustNewCustomElementsCounterVec(reg prometheus.Registerer) *prometheus.CounterVec {
+	return promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+		Name: "my_app_custom_elements_total",
+		Help: "Custom counter metric for my app counting important elements. It serves as an example of a very important metric that everyone is using.",
+		// Unit: "elements" // TODO(bwplotka): Add Unit as one of the supported options.
```

Comment:
Not strictly related to this PR: have you already thought how we should add Unit support?

Reply:
Yup, we add a field and put that in the UNIT field I presume (but it's not mandatory). So some part of your initial PR if you want to get that done 💪🏽